inst/builtin/bs5/shiny/_rules.scss

```
@@ -175,6 +175,20 @@ $bslib-checkbox-radio-margin-right: 0.35em !default;
 }
 
 
+// Aligns the bottom-border of nav links (i.e., underline) 
+// with the bottom-border of the navbar
+.navbar:has(.nav-underline) {
+  --#{$prefix}navbar-padding-y: 0;
+  .navbar-header {
+    padding-top: $navbar-padding-y;
+    padding-bottom: $navbar-padding-y;
+  }
+  .nav-link {
+    padding-top: $navbar-padding-y * 2;
+    padding-bottom: $navbar-padding-y * 2;
+  }
+}
```

I think you might be able to use negative margins to avoid the `:has()` and to avoid having to move the padding around:

```suggestion
.navbar .nav-underline .nav-link {
  padding-bottom: calc(var(--#{$prefix}navbar-nav-link-padding-y, #{$navbar-padding-y}) * 2);
  margin-bottom:  calc(var(--#{$prefix}navbar-nav-link-padding-y, #{$navbar-padding-y}) * -1);
}
```

or with just the Sass variables

```suggestion
.navbar .nav-underline .nav-link {
  padding-bottom: $navbar-padding-y * 2;
  margin-bottom:  $navbar-padding-y * -1;
}
```

---

Nice catch, thanks!